doc/developer/arrangements.md

```diff
@@ -60,15 +60,130 @@ Some of these input arrangements could be shared, but most likely the intermedia
 
 ---
 
-Our `Reduce` operator builds a non-trivial dataflow subgraph (which may be subject to change).
-Informally, for aggregation we build its own dataflow.
-That dataflow may begin with a `reduce` operator to implement the `distinct` keyword.
-Then there is either
-    * a single `reduce` operator (for sums, counts, any, all), or
-    * a sequence of 16 `reduce` operators (for min and max), or
-    * a single `reduce` operator (for `jsonbagg` and others).
-Finally, last `reduce` collects each of the aggregates into one record for each key.
-There are special cases when there are few (zero or one) reductions.
+Our `Reduce` operator builds a non-trivial dataflow subgraph.
+
+As of v0.7.1, the dataflow subgraph depends on 1) what types of aggregations are
+involved and 2) the type of input for the reduce.
+
+### Distinct (no aggregations)
+
+Creates two arrangements, each with row count `<number of groups>` and row width
+`<width of key>`.
+
+```
+In -> A -> A -> Out
+```
+
+### Only accumulable aggregations (sum/counts/any/all):
+
+Creates two arrangements, each with row count `<number of groups>` and row width
+`<width of key> + <width of aggregation columns>`.
+
+```
+In -> A -> A -> Out
+```
+
+### Only hierarchical aggregations (min/max) on an append-only input:
+
+Same as the "Only accumulable aggregations" case above.
+
+### Only hierarchical aggregations (min/max) on other inputs:
+
+Hierarchical aggregations on inputs that are not append-only have
+`ceil(log_16(expected_group_size))` stages, which are intended to filter the
+number of candidate rows by a factor of 16 per stage. If an
+`expected_group_size` has not been supplied via query hint, the default value
+is 4 billion.
+
+Each stage involves two arrangements.
+The first arrangement of the final stage is supposed to contain the top 16 rows
+for each group.
+The first arrangement of the second to last stage is supposed to contain the top
+16^2 = 256 rows for each group.
+The first arrangement of the third to last stage is supposed to contain the top
+16^3 = 4096 rows for each group.
+... and so on. Note that the first arrangement of the first stage will always
+contain all rows from the input.
+
+The second arrangement of the final stage is supposed to contain the top row
```

Comment:
I think theres a simpler description here -- "The output arrangments for all but the last stage of the reduction contain the inverse of the expected output, e.g. the rows which are _not_ the max or min so that we can bound the amount of memory spent on output arrangements to be at most the full size of the input across all stages"